pyproject.toml

```diff
@@ -0,0 +1,46 @@
+[project]
+name = "pastas"
+version = "0.22.0b"
+description = "Python package to perform time series analysis of hydrological time series."
+license = { file="LICENSE" }
+readme = "README.rst"
+authors = [ { name = "Collenteur et al. 2019" } ]
+maintainers = [
+{name="R.A. Collenteur", email="raoulcollenteur@gmail.com"},
+{name="M. Bakker", email="markbak@gmail.com"},
+{name="R. Calje", email="r.calje@artesia-water.nl"},
+{name="F. Schaars", email="f.schaars@artesia-water.nl"},
+{name="D.A. Brakenhoff", email="d.brakenhoff@artesia-water.nl"},
+{name="O. Ebbens", email="o.ebbens@artesia-water.nl"},
+{name="M.A. Vonk", email="m.vonk@artesia-water.nl"}
+]
+
+requires-python = ">=3.7"
+dependencies = [
+"numpy>=1.17",
+"matplotlib>=3.1",
+"pandas>=1.1",
+"scipy>=1.3"
+]
+
+keywords = ["hydrology", "groundwater", "timeseries", "analysis"]
+classifiers = [
+    'Development Status :: 4 - Beta',
+    'Intended Audience :: Science/Research',
+    'Intended Audience :: Other Audience',
+    'License :: OSI Approved :: MIT License',
+    'Programming Language :: Python :: 3 :: Only',
+    'Programming Language :: Python :: 3.7',
+    'Programming Language :: Python :: 3.8',
+    'Programming Language :: Python :: 3.9',
+    'Programming Language :: Python :: 3.10',
+    'Topic :: Scientific/Engineering :: Hydrology',
+]
+
+[project.urls]
+homepage = "https://pastas.dev"
+repository = "https://github.com/pastas/pastas"
+documentation = "https://pastas.readthedocs.io/en/latest/"
+
+[project.optional-dependencies]
+full = ["numba >= 0.51", "lmfit >= 1.0.0"]
```

Comment:
add custom installation options here (see flopy for example):
- docs (sphinx stuff)
- testing (pytest, pytest-cov, etc.)
- develop (everything?)

That way we can move all requirements for specific tasks to this file and get rid of all the .txt files.

Reply:
I added the packages to the pyproject.toml. However, I have not yet updated the installation of these packages for the read the docs from pyproject.toml.